v2/index/index.go

```diff
@@ -41,22 +35,22 @@ type (
 
 	// Index provides an interface for looking up byte offset of a given CID.
 	Index interface {
-		Codec() Codec
+		Codec() multicodec.Code
 		Marshal(w io.Writer) error
 		Unmarshal(r io.Reader) error
 		Get(cid.Cid) (uint64, error)
 		Load([]Record) error
 	}
 )
 
-// BuildersByCodec holds known index formats
-// TODO: turn this into a func before the final release?
-var BuildersByCodec = map[Codec]Builder{
-	IndexHashed:       mkHashed,
-	IndexSorted:       mkSorted,
-	IndexSingleSorted: mkSingleSorted,
-	IndexGobHashed:    mkGobHashed,
-	IndexInsertion:    mkInsertion,
+// NewFromCodec constructs a new index corresponding to the given codec.
+func NewFromCodec(codec multicodec.Code) (Index, error) {
```

Comment:
I wonder if just `New` would work. The parameter is a `multicodec.Code`, so it's pretty evident that a codec is used.

And `index.NewFromCodec(multicodec.IndexSorted)` feels repetitive, whereas `index.New(multicodec.IndexSorted)` isn't and still works well.